### DISCUSSION TITLE
Okhsv color saturation

### DISCUSSION THREAD

**Discussion by ad-on-is** [awawa-dev/HyperHDR#759] [General]
I just switched from hyperion to hyperhdr and I'm amazed by the speed and performance.

However, one thing I noticed is that the colors are a bit dull. Hyperion had this setting called Brightness/saturation gain, and they're using something called okhsv color calibration.

Would it be possible to have something similar?

https://github.com/hyperion-project/hyperion.ng/pull/1477

**Comment by awawa-dev** [comment]
Hi HyperHDR had from the beginning HSL customisation, just switch to HyperHDR classic calibration in the processing tab. Okhsv is not planned because it brings completely nothing important to LED processing.

**Comment by ad-on-is** [comment]
I switched to classic, but when playing with the settings, the colors are either dimmed and saturated or bright and "white-ish".

**Comment by awawa-dev** [comment]
And the video source is OK in the live preview? SDR? HDR? LED type and driver settings? These are aspect you need to focus. HSL or even okhsv is a placebo.

**Reply by ad-on-is** [reply]
Well it didn't look like a placebo to me. I could clearly see the difference. The setup was the same.

PiCap -> HyperHDR -> Led strip (wifi/udp). On the led-strip I tried WLED and hyperion controller for ESP.

Also, I forgot to mention that I use framebuffer as source (LG TV, PiCap) not a screen grabber.

**Reply by awawa-dev** [reply]
The change will be visible, but the use of these transformations for the signal (and after averaging... complete nonsense, at least from the theory side) has absolutely no sense and does not improve it. If you adjust one color channel to your liking, the subsequent channels will become unsatisfactory. If you start correcting them, you will break something you have already set up earlier, not to mention the color of the secondary ones.
And what LED type?

**Reply by awawa-dev** [reply]
BTW Hyperion is using different, hyper-candy gamma in the processing that causes to loose details but brings saturation to first plan. You can increase gamma settings in HyperHDR from default 1.5 to 2.2. They the effect will be the same.

**Reply by ad-on-is** [reply]
Leds are... WS281x

LoL, I know what you mean by adjusting and breaking. I wrote my own ambilight library for old Samsung TVs a few years ago. Back than I used solid colors for testing and did a bunch of if-else statements to make it look somewhat even 😅

This might sound stupid, but I really don't care that much about color accuracy as much as I care about brigtness and saturation. i.e. if there's an image with brown colors, I'd rather have some brighter "orange" on the LEDs, if you know what I mean. And I have the feeling that okhsv does exactly something like that.

I haven't been in the field for a while now and just recently repurposed my old LEDs on my new TV with hyperion, but I wasn't satisfied with the performance so I stumbled across HyperHDR, and now I'm back in business.

**Reply by awawa-dev** [reply]
I already told you that default HyperHDR settings are different than Hyperion. Change gamma first. That's the main but not only difference.

**Reply by ad-on-is** [reply]
I tried playing with gamma too, but I didn't notice much of a difference.

**Reply by awawa-dev** [reply]
If you still don't like it, you can import okhsv into LUT, then it will be even better because it will be applied at the pixel level and not their average. Additionally, you can import any other transformations this way. This is a much more flexible method that does not have to be limited to anything. That's all.

**Reply by ad-on-is** [reply]
Wait? I can apply LUTs to image processing that comes from a framebuffer? Didn't know that. I thought this was just for screen grabbers and HDR-SDR conversion, etc

**Reply by awawa-dev** [reply]
Yes, you can enable it in the flatbuffer settings.
